Dart/Sorting/Quicksort.dart

```diff
@@ -0,0 +1,65 @@
+// program to perform quick sort
+import 'dart:io';
+import 'dart:core';
+
+// function to swap two elements
+void swap(List array_1, int a, int b) {
+  int temp = array_1[a];
+  array_1[a] = array_1[b];
+  array_1[b] = temp;
+}
+
+// function to perform partition
+int partition(List array_1, int p, int r) {
+  int i = p - 1;
+  int pivot = array_1[r];
+  for (int j = p; j < r; j++) {
+    if (array_1[j] <= pivot) {
+      i++;
+      swap(array_1, i, j);
+    }
+  }
+  swap(array_1, i + 1, r);
+  return i + 1;
+}
+
+// function to perform the quicksort
+void quicksort(List array_1, int p, int r) {
+  if (p < r) {
+    int q = partition(array_1, p, r);
+    quicksort(array_1, p, q - 1);
+    quicksort(array_1, q + 1, r);
+  }
+}
+
+// Driver of Program
+void main() {
+  // scanning the size of the array
+  stdout.write("Enter the size of the array: ");
+  int size = int.parse(stdin.readLineSync()!);
```

Comment:
Hi @bishtanuj,
I am not a Dart expert but can you tell what is `!` here at the end is being used for?
As it throws error on an online compiler, removing it works fine

Reply:
Hi! It may be working properly on online platform but as I use VS Code, in which, not using this, shows error. It is a syntax for taking an input.

Reply:
Oh okay, probably it's for null check